Modules/_sqlite/util.c

```diff
@@ -28,15 +28,10 @@ int pysqlite_step(sqlite3_stmt* statement, pysqlite_Connection* connection)
 {
     int rc;
 
-    if (statement == NULL) {
-        /* this is a workaround for SQLite 3.5 and later. it now apparently
-         * returns NULL for "no-operation" statements */
-        rc = SQLITE_OK;
-    } else {
-        Py_BEGIN_ALLOW_THREADS
-        rc = sqlite3_step(statement);
-        Py_END_ALLOW_THREADS
-    }
+    Py_BEGIN_ALLOW_THREADS
+    /* returns SQLITE_MISUSE if called with a NULL argument */
```

Comment:
I think the comment makes the reader ask more questions rather than answer them (e.g. why we only mention about SQLITE_MISUSE?) I'd prefer removing it. The sqlite3 documentation is pretty clear about the function's return values.

Reply:
Good point. Removed in acd739a. Thanks.